### PR TITLE
ci: added dependabot.yml #1091

### LINKED ISSUE #1091: ✨ [Feature request]: Add dependabot gh action
### Is your feature request related to a problem? Please describe.

Dependabot is a tool that helps automate the process of keeping your project's dependencies up-to-date. Adding Dependabot to your project can have several benefits

### Describe the solution you'd like.

We can add a simple gh action 

### Describe alternatives you've considered.

N/A

### Screenshot / Screenshare

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by priyankarpal**
# Description
- Added dependabot gh action

- Fixes #1091 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
It's a gh action. It will update all gh actions if an update is needed.  

# Checklist:

- [ ] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Screenshots or example output

### CODE REVIEW

**Comment by Sachin-chaurasiya** [comment]
Hello @priyankarpal , Thanks for the PR, It would be great if you share some examples of how it will be working and what will be the benefits of adding this.

**Comment by priyankarpal** [comment]
> Hello @priyankarpal, Thanks for the PR, It would be great if you share some examples of how it will be working and what will be the benefits of adding this.

Dependabot is a tool that helps you keep your software dependencies up-to-date by automatically checking for updates and creating pull requests to update them.

**Comment by Sachin-chaurasiya** [comment]
> > Hello @priyankarpal, Thanks for the PR, It would be great if you share some examples of how it will be working and what will be the benefits of adding this.
> 
> Dependabot is a tool that helps you keep your software dependencies up-to-date by automatically checking for updates and creating pull requests to update them.

Will it also mention the vulnerabilities?

**Comment by priyankarpal** [comment]
> > > Hello @priyankarpal, Thanks for the PR, It would be great if you share some examples of how it will be working and what will be the benefits of adding this.
> > 
> > 
> > Dependabot is a tool that helps you keep your software dependencies up-to-date by automatically checking for updates and creating pull requests to update them.
> 
> Will it also mention the vulnerabilities?

No, this code does not check vulnerabilities directly.

This code appears to update the package ecosystem for GitHub Actions and schedule a daily update. However, it does not specify any specific tools or methods for checking vulnerabilities in the packages used in the repository or in the GitHub Actions workflow.

To check for vulnerabilities, you will need to use additional tools and services that can scan your code for known vulnerabilities in the packages you are using.

**Comment by Sachin-chaurasiya** [comment]
> > > > Hello @priyankarpal, Thanks for the PR, It would be great if you share some examples of how it will be working and what will be the benefits of adding this.
> > > 
> > > 
> > > Dependabot is a tool that helps you keep your software dependencies up-to-date by automatically checking for updates and creating pull requests to update them.
> > 
> > 
> > Will it also mention the vulnerabilities?
> 
> No, this code does not check vulnerabilities directly.
> 
> This code appears to update the package ecosystem for GitHub Actions and schedule a daily update. However, it does not specify any specific tools or methods for checking vulnerabilities in the packages used in the repository or in the GitHub Actions workflow.
> 
> To check for vulnerabilities, you will need to use additional tools and services that can scan your code for known vulnerabilities in the packages you are using.

@priyankarpal , It would be great to have some working examples.

**Comment by priyankarpal** [comment]
> > > > > Hello @priyankarpal, Thanks for the PR, It would be great if you share some examples of how it will be working and what will be the benefits of adding this.
> > > > 
> > > > 
> > > > Dependabot is a tool that helps you keep your software dependencies up-to-date by automatically checking for updates and creating pull requests to update them.
> > > 
> > > 
> > > Will it also mention the vulnerabilities?
> > 
> > 
> > No, this code does not check vulnerabilities directly.
> > This code appears to update the package ecosystem for GitHub Actions and schedule a daily update. However, it does not specify any specific tools or methods for checking vulnerabilities in the packages used in the repository or in the GitHub Actions workflow.
> > To check for vulnerabilities, you will need to use additional tools and services that can scan your code for known vulnerabilities in the packages you are using.
> 
> @priyankarpal , It would be great to have some working examples.

Dependabot will automatically create a PR if there are any updates needed. You can check the example of my project 

image

PR link: https://github.com/priyankarpal/ProjectsHut/pull/92

**Comment by Sachin-chaurasiya** [comment]
> > > > > > Hello @priyankarpal, Thanks for the PR, It would be great if you share some examples of how it will be working and what will be the benefits of adding this.
> > > > > 
> > > > > 
> > > > > Dependabot is a tool that helps you keep your software dependencies up-to-date by automatically checking for updates and creating pull requests to update them.
> > > > 
> > > > 
> > > > Will it also mention the vulnerabilities?
> > > 
> > > 
> > > No, this code does not check vulnerabilities directly.
> > > This code appears to update the package ecosystem for GitHub Actions and schedule a daily update. However, it does not specify any specific tools or methods for checking vulnerabilities in the packages used in the repository or in the GitHub Actions workflow.
> > > To check for vulnerabilities, you will need to use additional tools and services that can scan your code for known vulnerabilities in the packages you are using.
> > 
> > 
> > @priyankarpal , It would be great to have some working examples.
> 
> Dependabot will automatically create a PR if there are any updates needed. You can check the example of my project
> 
> image
> 
> PR link: [priyankarpal/ProjectsHut#92](https://github.com/priyankarpal/ProjectsHut/pull/92)

Thanks for sharing the example, looks good.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
+updates:
+  # Update GitHub Actions
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (Sachin-chaurasiya) [inline .github/dependabot.yml:7]: Can we make it weekly?

**Review by Sachin-chaurasiya — COMMENTED** [review]
Left a few comments, thanks

File: .github/dependabot.yml (hunk 1 above)
Comment (priyankarpal) [reply .github/dependabot.yml:7]: yeah sure

**Review by Sachin-chaurasiya — APPROVED** [review]
Looks Good 👍

**Review by atapas — APPROVED** [review]

**Pull request merged**